### ISSUE TITLE
Should have ability to choose hostname when sending.

### ISSUE THREAD

**Issue by peterhgombos** [gophish/gophish#532]
I've met a system that rejects the emails due to the HELO message in the SMTP exchange being set to only the hostname, not an FQDN. 

The code in question is here:

https://github.com/gophish/gophish/blob/master/worker/worker.go#L81

There are a few different ways I can imagine the fix, either setting the hostname to the FQDN of the server  (hostname --fqdn), or letting the user set the hostname, either in the config of the installation or per campaign. If I could choose, I would let the user set a default hostname when installing, and then have an option to change it before sending a new campaign. 

This can also be an option in the sending profile.

What do you think?

**Comment by jordan-wright** [comment]
That's surprising that the SMTP server would reject emails if a full hostname isn't provided.

I'll be thinking about the best way to do this. From what I've seen, this is a rare scenario.

Thanks for the heads up!

**Comment by peterhgombos** [comment]
I agree that it's weird, I ended up getting our instance white listed. This might be part of a bigger issue, that is to get more control over the email headers when sending. Some setting page allowing full control if you want it, perhaps?

**Comment by syserr0r** [comment]
This applies to our situation as well, but because there is a mailserver for one of our subdomains on the machine we are running from and the primary mailserver thinks it is impersonating the mailserver.

Being able to specify a different domain name (the host has multiple) would be great!

**Comment by TriYop** [comment]
It seems that outlook 2019 does some checks and raises warnings on incoming mails with unexpected domains (even if SPF, DKIM authentication pass).

Allowing to set domain would greatly help.

**Comment by phillycheeze** [comment]
This would also be helpful for using the GSuite SMTP relay. They require the domain name to be in the HELO command. 
https://support.google.com/a/answer/2956491

Snippet from documentation:

> For the Any address option to work properly, you must configure your mail server either to use SMTP AUTH to identify the sending domain or to present one of your domain names in the HELO or EHLO command. See the instructions below for configuring your specific server type. You must also configure your mail server in one of these ways if you send messages from a domain you don't own (such as yahoo.com), or if you send messages with an empty envelope-from, such as non-delivery reports or vacation “out of office”

**Comment by martinvonwittich** [comment]
> The code in question is here:
> 
> https://github.com/gophish/gophish/blob/master/worker/worker.go#L81

Press `y` on GitHub to get a permalink: https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files

In this case, the correct link is probably https://github.com/gophish/gophish/blob/19021b6c91a919dc68e67309f44f19d928d30cdb/worker/worker.go#L81

I would like the ability to change the hostname that GoPhish is using as the HELO because the actual hostname kinda gives away that the mail is sent from an internal company host. I cannot change the actual hostname of the machine because that would break our internal autoupdate system.
